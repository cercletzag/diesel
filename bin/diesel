#!/bin/sh
cd diesel_cli && cargo run -- $@
