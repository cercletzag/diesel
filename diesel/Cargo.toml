[package]
name = "diesel"
version = "0.4.1"
authors = ["Sean Griffin <sean@seantheprogrammer.com>"]
license = "MIT OR Apache-2.0"
description = "A safe, extensible ORM and Query builder"
documentation = "http://sgrif.github.io/diesel/diesel/index.html"
homepage = "https://github.com/sgrif/diesel"
repository = "https://github.com/sgrif/diesel"
keywords = ["orm", "database", "postgres", "postgresql", "sql"]

[dependencies]
libc = "0.2.*"
pq-sys = { version = "^0.2.0", optional = true }
libsqlite3-sys = { version = "^0.4.0", optional = true }
byteorder = "0.3.*"
quickcheck = { git = "https://github.com/BurntSushi/quickcheck.git", optional = true }
chrono = { version = "^0.2.17", optional = true }

[dev-dependencies]
quickcheck = { git = "https://github.com/BurntSushi/quickcheck.git" }
dotenv = "^0.6.0"
tempdir = "^0.3.4"

[features]
default = ["postgres"]
unstable = []
large-tables = []
postgres = ["pq-sys"]
sqlite = ["libsqlite3-sys"]
