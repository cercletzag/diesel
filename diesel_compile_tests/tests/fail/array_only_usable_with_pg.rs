extern crate diesel;

use diesel::dsl::*;
use diesel::*;

fn main() {
    let connection = SqliteConnection::establish("").unwrap();
    select(array((1,))).get_result::<Vec<i32>>(&connection);

    let connection = MysqlConnection::establish("").unwrap();
    select(array((1,))).get_result::<Vec<i32>>(&connection);
}
