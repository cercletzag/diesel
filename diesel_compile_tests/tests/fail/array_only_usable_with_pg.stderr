error[E0271]: type mismatch resolving `<diesel::SqliteConnection as diesel::Connection>::Backend == diesel::pg::Pg`
 --> $DIR/array_only_usable_with_pg.rs:8:25
  |
8 |     select(array((1,))).get_result::<Vec<i32>>(&connection);
  |                         ^^^^^^^^^^ expected struct `diesel::sqlite::Sqlite`, found struct `diesel::pg::Pg`
  |
  = note: required because of the requirements on the impl of `diesel::query_dsl::LoadQuery<diesel::SqliteConnection, std::vec::Vec<i32>>` for `diesel::query_builder::SelectStatement<(), diesel::query_builder::select_clause::SelectClause<diesel::pg::expression::array::ArrayLiteral<(diesel::expression::bound::Bound<diesel::sql_types::Integer, i32>,), diesel::sql_types::Integer>>>`

error[E0277]: the trait bound `diesel::sqlite::Sqlite: diesel::sql_types::HasSqlType<diesel::sql_types::Array<diesel::sql_types::Integer>>` is not satisfied
 --> $DIR/array_only_usable_with_pg.rs:8:25
  |
8 |     select(array((1,))).get_result::<Vec<i32>>(&connection);
  |                         ^^^^^^^^^^ the trait `diesel::sql_types::HasSqlType<diesel::sql_types::Array<diesel::sql_types::Integer>>` is not implemented for `diesel::sqlite::Sqlite`
  |
  = help: the following implementations were found:
            <diesel::sqlite::Sqlite as diesel::sql_types::HasSqlType<diesel::sql_types::BigInt>>
            <diesel::sqlite::Sqlite as diesel::sql_types::HasSqlType<diesel::sql_types::Binary>>
            <diesel::sqlite::Sqlite as diesel::sql_types::HasSqlType<diesel::sql_types::Bool>>
            <diesel::sqlite::Sqlite as diesel::sql_types::HasSqlType<diesel::sql_types::Date>>
          and 8 others
  = note: required because of the requirements on the impl of `diesel::expression::QueryMetadata<diesel::sql_types::Array<diesel::sql_types::Integer>>` for `diesel::sqlite::Sqlite`
  = note: required because of the requirements on the impl of `diesel::query_dsl::LoadQuery<diesel::SqliteConnection, std::vec::Vec<i32>>` for `diesel::query_builder::SelectStatement<(), diesel::query_builder::select_clause::SelectClause<diesel::pg::expression::array::ArrayLiteral<(diesel::expression::bound::Bound<diesel::sql_types::Integer, i32>,), diesel::sql_types::Integer>>>`

error[E0271]: type mismatch resolving `<diesel::MysqlConnection as diesel::Connection>::Backend == diesel::pg::Pg`
  --> $DIR/array_only_usable_with_pg.rs:11:25
   |
11 |     select(array((1,))).get_result::<Vec<i32>>(&connection);
   |                         ^^^^^^^^^^ expected struct `diesel::mysql::Mysql`, found struct `diesel::pg::Pg`
   |
   = note: required because of the requirements on the impl of `diesel::query_dsl::LoadQuery<diesel::MysqlConnection, std::vec::Vec<i32>>` for `diesel::query_builder::SelectStatement<(), diesel::query_builder::select_clause::SelectClause<diesel::pg::expression::array::ArrayLiteral<(diesel::expression::bound::Bound<diesel::sql_types::Integer, i32>,), diesel::sql_types::Integer>>>`

error[E0277]: the trait bound `diesel::mysql::Mysql: diesel::sql_types::HasSqlType<diesel::sql_types::Array<diesel::sql_types::Integer>>` is not satisfied
  --> $DIR/array_only_usable_with_pg.rs:11:25
   |
11 |     select(array((1,))).get_result::<Vec<i32>>(&connection);
   |                         ^^^^^^^^^^ the trait `diesel::sql_types::HasSqlType<diesel::sql_types::Array<diesel::sql_types::Integer>>` is not implemented for `diesel::mysql::Mysql`
   |
   = help: the following implementations were found:
             <diesel::mysql::Mysql as diesel::sql_types::HasSqlType<diesel::sql_types::BigInt>>
             <diesel::mysql::Mysql as diesel::sql_types::HasSqlType<diesel::sql_types::Binary>>
             <diesel::mysql::Mysql as diesel::sql_types::HasSqlType<diesel::sql_types::Bool>>
             <diesel::mysql::Mysql as diesel::sql_types::HasSqlType<diesel::sql_types::Date>>
           and 15 others
   = note: required because of the requirements on the impl of `diesel::expression::QueryMetadata<diesel::sql_types::Array<diesel::sql_types::Integer>>` for `diesel::mysql::Mysql`
   = note: required because of the requirements on the impl of `diesel::query_dsl::LoadQuery<diesel::MysqlConnection, std::vec::Vec<i32>>` for `diesel::query_builder::SelectStatement<(), diesel::query_builder::select_clause::SelectClause<diesel::pg::expression::array::ArrayLiteral<(diesel::expression::bound::Bound<diesel::sql_types::Integer, i32>,), diesel::sql_types::Integer>>>`
